src/pudl/workspace/datastore.py

```diff
@@ -167,8 +167,11 @@ def _validate_datapackage(self, datapackage_json: dict):
 
         Throws ValueError if invalid.
         """
-        dp = datapackage.Package(datapackage_json)
-        if not dp.valid:
+        # TODO (daz): when we upgrade to frictionless>=5.0, this will be:
+        # dp = frictionless.Package.validate_descriptor(datapackage_json)
+        # if not dp.valid:
+        dp = frictionless.Package(datapackage_json)
+        if not dp.metadata_validate():
```

Comment:
I could be remembering wrong, but I'm suspicious that this validation may **only** checking the validity of the topmost metadata elements, and not descending into any of the referenced resources to check whether they are also valid. If that's true, it would be informative to try validating the resources and see whether they also pass.

I think a lot of the rest of this class is also doing work that frictionless should really be doing, like validating the file sizes, MD5 checksums and potentially paths.

Reply:
Yes, this is not properly checking the referenced resources. But, since the referenced resources aren't defined with the correct `{"dialect": {"sql": {"table": "blah"}}}` format, and frictionless 4 doesn't know what SQL files are anyways, I think this is the best we can do at this point...

Reply:
In the datastore use case, we're using generic [data resources](https://specs.frictionlessdata.io/data-resource/) which don't have any schema / dialect attributes or know anything about the internal structure of the data. They're just files of any kind, with a name, description, path, filetype, size, hash, and potentially other freeform metadata. This is in contrast to the [tabular data resources](https://specs.frictionlessdata.io/tabular-data-resource/) which are intended to represent tables and can describe much more of the expected internal structure of the resource.

Since we're just trying to bump to the new version I think addressing missing validation step this is well outside the scope of this PR, but I think we can validate these resources and should try to get that working in a future PR so we don't have so much of our own infrastructure paralleling stuff that Frictionless can/should do.

Reply:
Ah, right, since we're not downloading any actual SQL resources into $PUDL_INPUT - that only affects our fancy `fercX_xbrl_datapackage.json`s.

That being said, I've some write-up of what it would take to get all this stuff into Frictionless in #3293 .